Commonality Analysis Code.Rmd

```diff
@@ -0,0 +1,258 @@
+cope_ca_data <- cope_ca_data %>% 
+  filter(condition == 1 | condition == 2) %>%
+  filter(pi_dropout == 0)
+
+#Change numeric-coded and integer-coded categorical variables into factors and make cope_ca_data a data.frame for missForest to work
+cope_ca_data <- as.data.frame(cope_ca_data) %>% 
+  mutate(b_drs_month = factor(b_drs_month),
+    f1_drs_month = factor(f1_drs_month),
+     across(starts_with("b_dem_gender"), factor),
+    across(starts_with("b_dem_sex"), factor),
+    across(starts_with("b_dem_rom"), factor),
+    across(starts_with("b_dem_race"), factor))
+
+#Since missForest can only work with categorical variables with a maximum of 53 levels, need to lump together factors. Decided to lump those with less than 10 responses.
+str(cope_ca_data)
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex = fct_lump_min(b_dem_sex, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_3_text = fct_lump_min(b_dem_sex_3_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_gender = fct_lump_min(b_dem_gender, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_gender_16_text = fct_lump_min(b_dem_gender_16_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_orientation = fct_lump_min(b_dem_orientation, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_orientation_7_text = fct_lump_min(b_dem_orientation_7_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_experience = fct_lump_min(b_dem_sex_experience, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_experience_10_text = fct_lump_min(b_dem_sex_experience_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_attract = fct_lump_min(b_dem_sex_attract, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_sex_attract_10_text = fct_lump_min(b_dem_sex_attract_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_rom_attract = fct_lump_min(b_dem_rom_attract, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_rom_attract_10_text = fct_lump_min(b_dem_rom_attract_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_race = fct_lump_min(b_dem_race, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_race_7_text = fct_lump_min(b_dem_race_7_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_language = fct_lump_min(b_dem_language, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_language_10_text = fct_lump_min(b_dem_language_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_zip_code = fct_lump_min(b_dem_zip_code, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_grade = fct_lump_min(b_dem_grade, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_grade_9_text = fct_lump_min(b_dem_grade_9_text, 10)) 
+str(cope_ca_data)
+```
+
+```{r data imputation}
+set.seed(12345) #use the same seed number each time to make the code reproducible (the seed number is just the starting point used in the generation of a sequence of random numbers in the random forest)
+

```

Comment:
What else would you recommend cutting? Trying to understand the balance of what info is helpful for imputation and what is irrelevant

Reply:
I think you just have to go off of your intuition! How useful do you think it would be for predicting someone's mental health to know whether they're cis or trans? Probably very helpful! What about trans versus pan versus demi etc.? Probably less helpful (I would think anyway). Any time you have a lot of categories and they don't seem that useful they're more likely to confuse the model than anything else. But just go with your gut.